SpiceSharp/Algebra/Solve/ComplexSolver.cs

```diff
@@ -0,0 +1,333 @@
+
+            // Get the diagonal
+            element = Matrix.GetDiagonalElement(1);
+            if (element.Value.Equals(0))
+                return false;
+
+            // pivot = 1 / pivot
+            element.Value = Inverse(element.Value);
+
+            // TODO: maybe we should cache this
+            MatrixElement<Complex>[] dest = new MatrixElement<Complex>[Matrix.Size + 1];
+
+            // Start factorization
+            Complex mult;
+            for (int step = 2; step <= Matrix.Size; step++)
+            {
+                // Scatter
+                element = Matrix.GetFirstInColumn(step);
+                while (element != null)
+                {
+                    dest[element.Row] = element;
+                    element = element.Below;
+                }
+
+                // Update column
+                column = Matrix.GetFirstInColumn(step);
+                while (column.Row < step)
+                {
+                    element = Matrix.GetDiagonalElement(column.Row);
+
+                    // Mult = dest[row] / pivot
+                    mult = dest[column.Row].Value * element.Value;
+                    dest[column.Row].Value = mult;
+                    while ((element = element.Below) != null)
+                    {
+                        // dest[element.Row] -= mult * element
+                        dest[element.Row].Value -= mult * element.Value;
+                    }
+                    column = column.Below;
+                }
+
+                // Check for a singular matrix
+                element = Matrix.GetDiagonalElement(step);
+                if (element == null || element.Value.Equals(0.0))
+                    return false;
+                element.Value = Inverse(element.Value);
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Find the solution for a factored matrix and a right-hand-side
+        /// </summary>
+        /// <param name="solution">Solution vector</param>
+        public override void Solve(Vector<Complex> solution)
+        {
+            if (solution == null)
+                throw new ArgumentNullException(nameof(solution));
+
+            // TODO: Maybe we should cache intermediate
+            // Scramble
+            var intermediate = new Complex[Matrix.Size + 1];
+            var rhsElement = Rhs.First;
+            while (rhsElement != null)
+            {
+                intermediate[rhsElement.Index] = rhsElement.Value;
+                rhsElement = rhsElement.Next;
+            }
+
+            // Forward substitution
+            for (int i = 1; i <= Matrix.Size; i++)
+            {
+                Complex temp = intermediate[i];
+
+                // This step of the substitution is skipped if temp == 0.0
+                if (!temp.Equals(0.0))
+                {
+                    var pivot = Matrix.GetDiagonalElement(i);
+
+                    // temp = temp / pivot
```

Comment:
@svenboulanger  temp = temp * pivot ?

Reply:
The pivot is inverted during factorization. So `/ pivot` becomes `* pivot.Value`